cmd/boulder/main.go

```diff
@@ -60,57 +55,102 @@ func readAndValidateConfigFile(name, filename string) error {
+	for i, arg := range os.Args {
+		if arg == "--"+flag || arg == "-"+flag {
+			os.Args = append(os.Args[:i], os.Args[i+1:]...)
 			return
 		}
-		os.Args = os.Args[1:]
-		subcommand()
-	}, nil)
-	// TODO(#6763): Move this inside of main().
-	cmd.RegisterCommand("--list", func() {
-		for _, c := range cmd.AvailableCommands() {
-			if c != "boulder" && c != "--list" {
-				fmt.Println(c)
+	}
+}
+
+// getConfigPath returns the path to the config file if it was provided as a
+// command line flag. If the flag was not provided, it returns an empty string.
+func getConfigPath() string {
+	for i := 0; i < len(os.Args); i++ {
+		if os.Args[i] == "--config" || os.Args[i] == "-config" {
+			if i+1 < len(os.Args) {
+				return os.Args[i+1]
 			}
 		}
-	}, nil)
-	// TODO(#6763): Move this inside of main().
-	cmd.RegisterCommand("validate", func() {
+	}
+	return ""
+}
+
+var boulderUsage = fmt.Sprintf(`Usage: %s <subcommand> [flags]
+
+  Each boulder component has its own subcommand. Use --list to see
+  a list of the available components. Use <subcommand> --help to
+  see the usage for a specific component. 
+
+global flag(s):
+  --skip-config-validation/ -skip-config-validation:
+        Skip validation of the configuration file.
+`,
+	core.Command())
+
+func main() {
+	var subcommand func()
+	if core.Command() == "boulder" {
+		// The operator is running the boulder binary directly.
 		if len(os.Args) <= 1 {
-			fmt.Fprintf(os.Stderr, "Call with --help to list usage.\n")
-			os.Exit(1)
+			// No arguments passed.
+			fmt.Fprint(os.Stderr, boulderUsage)
+			return
 		}
-		list := flag.Bool("list", false, "List available components to validate configuration for.")
-		component := flag.String("component", "", "The name of the component to validate configuration for.")
-		configFile := flag.String("config", "", "The path to the configuration file to validate.")
-		flag.Parse()
 
-		if *list {
-			for _, c := range cmd.AvailableConfigValidators() {
+		if os.Args[1] == "--help" || os.Args[1] == "-help" {
+			// Help flag passed.
+			fmt.Fprint(os.Stderr, boulderUsage)
+			return
+		}
+
+		if os.Args[1] == "--list" || os.Args[1] == "-list" {
+			// List flag passed.
+			for _, c := range cmd.AvailableCommands() {
 				fmt.Println(c)
 			}
 			return
 		}
-		if *component == "" || *configFile == "" {
-			fmt.Fprintf(os.Stderr, "Must provide a configuration file to validate.\n")
+
+		subcommand = cmd.LookupCommand(os.Args[1])
+		if subcommand == nil {
+			fmt.Fprintf(os.Stderr, "Unknown subcommand '%s'.\n", os.Args[1])
 			os.Exit(1)
 		}
-		err := readAndValidateConfigFile(*component, *configFile)
+	}
+
+	configPath := getConfigPath()
+	if configPath != "" && !skipConfigValidationSpecified() {
+		// Config flag passed.
+		err := readAndValidateConfigFile(core.Command(), configPath)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error validating configuration: %s\n", err)
+			fmt.Fprintf(os.Stderr, "Error validating config file %q: %v\n", configPath, err)
```

Comment:
```suggestion
			fmt.Fprintf(os.Stderr, "Error validating config file %q: %s\n", configPath, err)
```

Generally better to use the %s output for errors.